src/main/scala/com/databricks/labs/overwatch/pipeline/Snapshot.scala

```diff
@@ -0,0 +1,124 @@
+package com.databricks.labs.overwatch.pipeline
+
+import com.databricks.labs.overwatch.utils.{CloneDetail, CloneReport, Config, Helpers, OverwatchScope, SparkSessionWrapper, WorkspaceDataset}
+import org.apache.log4j.Logger
+import com.databricks.labs.overwatch.env.{Database, Workspace}
+import org.apache.spark.sql.{Column, DataFrame, Dataset}
+import org.apache.log4j.{Level, Logger}
+
+
+object Snapshot extends SparkSessionWrapper {
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+  import spark.implicits._
+
+  /**
+   * Create a backup of the Overwatch datasets
+   *
+   * @param snapshotRootPath prefix of path target to send the snap
+   * @param cloneLevel       Deep or Shallow
+   * @param asOfTS           appends asOfTimestamp option to Delta reader to limit data on clone. This will only go back as
+   *                         far as the latest vacuum by design.
+   * @param excludes         Array of table names to exclude from the snapshot
+   *                         this is the table name only - without the database prefix
+   * @return
+   */
+  private[overwatch] def snap(
+              bronze: Bronze,
+              workspace: Workspace,
+              snapshotRootPath: String,
+              cloneLevel: String = "DEEP",
```

Comment:
This should be able to be overridden in the companion object apply method -- as of now, it's always default and unchangeable outside of Overwatch package.